src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -427,22 +446,21 @@ export class DashboardAppController {
               refreshDashboardContainer();
             });
 
-            const incomingState = embeddable
-              .getStateTransfer(scopedHistory())
-              .getIncomingEmbeddablePackage();
-            if (incomingState) {
-              if ('id' in incomingState) {
-                container.addOrUpdateEmbeddable<SavedObjectEmbeddableInput>(incomingState.type, {
-                  savedObjectId: incomingState.id,
-                });
-              } else if ('input' in incomingState) {
-                const input = incomingState.input;
-                delete input.id;
-                const explicitInput = {
-                  savedVis: input,
-                };
-                container.addOrUpdateEmbeddable<EmbeddableInput>(incomingState.type, explicitInput);
-              }
+            // If the incomingEmbeddable does not yet exist in the panels listing, create a new panel using the container's addEmbeddable method.
+            if (
+              incomingEmbeddable &&
+              (!incomingEmbeddable.embeddableId ||
+                !container.getInput().panels[incomingEmbeddable.embeddableId])
+            ) {
+              // TODO: Get rid of this, maybe by making the visualize embeddable also use the attributeService
+              const explicitInput =
+                incomingEmbeddable.type === 'lens' ||
```

Comment:
Same question as Peter - I am also not following this change. Why do you need to check if the embeddable type is Lens explicitly?

Reply:
I would be really happy to get rid of this before merging. The problem is that the lens embeddable is using the attributeService, which expects the 'by value' input to be in a key named `attributes`, whereas the visualize embeddable is looking for a key named `savedVis `. I was hoping that it would be possible to

Reply:
I was able to remove this line by making a small change in the visualize app, to pre-wrap the by value input in a `savedVis` key. I would still consider making the visualize embeddable use the `applicationService` soon, just for greater consistency,